src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs

```diff
@@ -179,15 +166,14 @@ public virtual void Close()
                 // We don't need to terminate if the test host has already terminated.
                 // If the host launch fails, please clean test host. 
                 // The upper bound for wait should be 100ms.
-                if ((this.testHostProcessId != -1 && !this.testHostExited.Wait(100)) || this.testHostLaunchError.Wait(0))
+                if (!this.testHostExited.Wait(100))
```

Comment:
`Clean` should be called irrespective of whether the test host has exited or not.